### DISCUSSION TITLE
Trouble enabling typescript language extra

### DISCUSSION THREAD

**Discussion by Jacobus-afk** [LazyVim/LazyVim#4290] [Q&A]
Hi

Before I create an issue, I just wanted to check if perhaps I'm not doing something wrong.

I added typescript as an extra by following the instructions [here](http://www.lazyvim.org/extras/lang/typescript)

After quitting and opening nvim, I checked in Mason if the lsp(vtsls) is installed.. no joy

So I tried to manually install it in `lua/plugins/`:

```lua title="lua/plugins/extend-mason"
return {
  "williamboman/mason.nvim",
  opts = {
    ensure_installed = {
       "vtsls",
    },
  },
}
```
After that, opening a typescript file, gives the following exception

`  Error  12:36:20 notify.error mason-lspconfig.nvim ...y/LazyVim/lua/lazyvim/plugins/extras/lang/typescript.lua:174: attempt to index field 'settings' (a nil value)`

Taking a look at line 174 in `LazyVim/lua/lazyvim/plugins/extras/lang/typescript.lua`

```lua
-- copy typescript settings to javascript
opts.settings.javascript =
  vim.tbl_deep_extend("force", {}, opts.settings.typescript, opts.settings.javascript or {})
```
Can someone please explain the above functionality. Do I need to install a javascript language plugin as well? Or is this a bona fide bug?

**Comment by dpetka2001** [comment]
How did you enable the Extra? Just type `:LazyExtras` and see if the Typescript Extra is under `Enabled Languages`. If not you did something wrong.

**Reply by Jacobus-afk** [reply]
Hi dpetka

I enabled it by adding `{ import = "lazyvim.plugins.extras.lang.typescript" }` to `lua/config/lazy.lua`

screenshot of my enabled languages in :LazyExtras

![image](https://github.com/user-attachments/assets/ea3326c4-4478-49a1-a0b6-1a2aa4a31b8c)

**Reply by dpetka2001** [reply]
I just tried with a default LazyVim installation and the `vtsls` gets installed correctly and also attached to the buffer. Following screencast

[Screencast 2024-08-12 11:14:03.webm](https://github.com/user-attachments/assets/0638308e-97af-4ca5-8560-951d7e18ff91)

**Reply by Jacobus-afk** [reply]
Thanks for the effort! That pretty much confirms it must be something in my config
